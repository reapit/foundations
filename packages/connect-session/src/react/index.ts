import React from 'react'
import { ReapitConnectHook, ReapitConnectSession } from '../types'
import { ReapitConnectBrowserSession } from '../browser'

export const useReapitConnect = (reapitConnectBrowserSession: ReapitConnectBrowserSession): ReapitConnectHook => {
  const [connectSession, setConnectSession] = React.useState<ReapitConnectSession | null>(null)

  React.useEffect(() => {
    const connectGetSession = async () => {
      const session = await reapitConnectBrowserSession.connectSession()
      if (session) {
        setConnectSession(session)
      }
    }

    connectGetSession()
  }, [])

  const connectAuthorizeRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectAuthorizeRedirect()
  }, [])

  const connectLoginRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLoginRedirect()
  }, [])

  const connectLogoutRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLogoutRedirect()
  }, [])

  const connectIsDesktop = reapitConnectBrowserSession.connectIsDesktop

  const connectHasSession = reapitConnectBrowserSession.connectHasSession

  return {
    connectSession,
    connectAuthorizeRedirect,
    connectLoginRedirect,
    connectLogoutRedirect,
    connectIsDesktop,
    connectHasSession,
  }
}
