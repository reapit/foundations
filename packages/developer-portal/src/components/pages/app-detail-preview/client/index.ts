export { default } from './app-detail'
