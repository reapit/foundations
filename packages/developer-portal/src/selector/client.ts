import { ReduxState } from '@/types/core'
import { selectLoginIdentity } from '@/selector/auth'
import { COGNITO_GROUP_DEVELOPER_EDITION } from '@/constants/api'

export const selectClientId = (state: ReduxState) => {
  return state?.auth?.loginSession?.loginIdentity?.clientId || ''
}

/**
 * Need get developer id to filter apps list if this user belong to
 * AgencyCloudDeveloperEdition group, if not just return null
 * refer to this ticket https://github.com/reapit/foundations/issues/1848
 */
export const selectDeveloperEditionId = (state: ReduxState) => {
  const loginIdentity = selectLoginIdentity(state)
  if (loginIdentity?.groups.includes(COGNITO_GROUP_DEVELOPER_EDITION)) {
    return state?.auth?.loginSession?.loginIdentity?.developerId || ''
  }
  return null
}

export const selectLoggedUserEmail = (state: ReduxState): string => {
  return state?.auth?.loginSession?.loginIdentity?.email || ''
}
