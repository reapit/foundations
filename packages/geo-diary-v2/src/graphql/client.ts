import ApolloClient, { InMemoryCache, Operation, defaultDataIdFromObject, IdGetterObj } from 'apollo-boost'
import { ErrorHandler, ErrorResponse } from 'apollo-link-error'
import { ApolloCache } from 'apollo-cache'
import typeDefs from './schema.graphql'
import resolvers from './resolvers'

export const generateRequest = (accessToken: string) => async (operation: Operation) => {
  operation.setContext({
    headers: {
      authorization: accessToken,
    },
  })
}

export const onError: ErrorHandler = ({ graphQLErrors, networkError }: ErrorResponse) => {
  if (graphQLErrors) {
    graphQLErrors.map(({ message, locations, path }) =>
      console.log(`[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`),
    )
  }
  if (networkError) console.log(`[Network error]: ${networkError}`)
}

export const dataIdFromObject = (object: IdGetterObj) => {
  // Will open when custom object ID for caching
  // switch (object.__typename) {
  //   // Custom keycache here!
  //   default:
  //     return defaultDataIdFromObject(object)
  // }
  return defaultDataIdFromObject(object)
}

const cache: ApolloCache<any> = new InMemoryCache({
  dataIdFromObject,
})

const clientState = {
  cache,
  defaults: {},
  resolvers,
  typeDefs,
}

export const getClient = (accessToken: string, uri: string) =>
  new ApolloClient({
    uri,
    onError,
    cache,
    request: generateRequest(accessToken),
    clientState,
  })

export default getClient
