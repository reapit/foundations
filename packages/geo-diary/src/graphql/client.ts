import ApolloClient, { InMemoryCache, Operation, defaultDataIdFromObject, IdGetterObj } from 'apollo-boost'
import { ErrorHandler, ErrorResponse } from 'apollo-link-error'
import { ApolloCache } from 'apollo-cache'
import typeDefs from './schema.graphql'
import resolvers from './resolvers'
import { notification } from '@reapit/elements'
import { ReapitConnectSession } from '@reapit/connect-session'

export const generateRequest = (session: ReapitConnectSession) => async (operation: Operation) => {
  const { loginIdentity, accessToken } = session
  operation.setContext({
    headers: {
      authorization: accessToken,
      'reapit-customer': `${loginIdentity.clientId}-${loginIdentity.userCode}`,
    },
  })
}

export const onError: ErrorHandler = ({ graphQLErrors, networkError }: ErrorResponse) => {
  if (graphQLErrors) {
    graphQLErrors.map(({ message, locations, path }) => {
      console.error(`[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`)
      const messageArr = message?.split('-')
      const LAST_INDEX = messageArr?.length - 1
      const messageNotIncludeTraceID = messageArr?.[LAST_INDEX]
      notification.error({
        message: messageNotIncludeTraceID,
        placement: 'bottomRight',
      })
    })
  }
  if (networkError) {
    console.error(`[Network error]: ${networkError}`)
    notification.error({
      message: networkError,
      placement: 'bottomRight',
    })
  }
}

export const dataIdFromObject = (object: IdGetterObj) => {
  // Will open when custom object ID for caching
  // switch (object.__typename) {
  //   // Custom keycache here!
  //   default:
  //     return defaultDataIdFromObject(object)
  // }
  return defaultDataIdFromObject(object)
}

const cache: ApolloCache<any> = new InMemoryCache({
  dataIdFromObject,
})

const clientState = {
  cache,
  defaults: {},
  resolvers,
  typeDefs,
}

export const getClient = (session: ReapitConnectSession, uri: string) =>
  new ApolloClient({
    uri,
    onError,
    cache,
    request: generateRequest(session),
    clientState,
  })

export default getClient
