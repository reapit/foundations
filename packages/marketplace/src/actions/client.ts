import { actionCreator } from '@/utils/actions'
import ActionTypes from '@/constants/action-types'
import { ClientAppSummary, ClientAppSummaryParams } from '@/reducers/client/app-summary'
import { AppDetailData } from '@/reducers/client/app-detail'
import { FetchAppByIdParams } from '@/services/apps'
import {
  PutWebComponentConfigParams,
  FetchWebComponentConfigParams,
  WebComponentConfigResult,
} from '@/services/web-component'
import { NegotiatorsResult } from '@/services/negotiators'

export const clientFetchAppSummary = actionCreator<ClientAppSummaryParams>(ActionTypes.CLIENT_FETCH_APP_SUMMARY)
export const clientFetchAppSummarySuccess = actionCreator<ClientAppSummary | undefined>(
  ActionTypes.CLIENT_FETCH_APP_SUMMARY_SUCCESS,
)
export const clientFetchAppSummaryFailed = actionCreator<string>(ActionTypes.CLIENT_FETCH_APP_SUMMARY_FAILED)
export const clientClearAppSummary = actionCreator<null>(ActionTypes.CLIENT_CLEAR_APP_SUMMARY)

// Client App Detail
export const clientFetchAppDetail = actionCreator<FetchAppByIdParams>(ActionTypes.CLIENT_FETCH_APP_DETAIL)
export const clientFetchAppDetailSuccess = actionCreator<AppDetailData>(ActionTypes.CLIENT_FETCH_APP_DETAIL_SUCCESS)
export const clientFetchAppDetailFailed = actionCreator<string>(ActionTypes.CLIENT_FETCH_APP_DETAIL_FAILED)

export const clientOpenWebComponentConfig = actionCreator<void>(ActionTypes.CLIENT_WEB_COMPONENT_CONFIG_OPEN)
export const clientCloseWebComponentConfig = actionCreator<void>(ActionTypes.CLIENT_WEB_COMPONENT_CONFIG_CLOSE)
export const clientFetchWebComponentConfig = actionCreator<FetchWebComponentConfigParams>(
  ActionTypes.CLIENT_FETCH_WEB_COMPONENT_CONFIG,
)
export const clientFetchWebComponentConfigSuccess = actionCreator<WebComponentConfigResult>(
  ActionTypes.CLIENT_FETCH_WEB_COMPONENT_CONFIG_SUCCESS,
)
export const clientPutWebComponentConfig = actionCreator<PutWebComponentConfigParams>(
  ActionTypes.CLIENT_PUT_WEB_COMPONENT_CONFIG,
)

export const clientFetchNegotiatorsSuccess = actionCreator<NegotiatorsResult>(
  ActionTypes.CLIENT_FETCH_NEGOTIATORS_SUCCESS,
)
