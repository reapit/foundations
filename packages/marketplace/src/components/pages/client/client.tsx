import * as React from 'react'
import { History } from 'history'
import { useSelector } from 'react-redux'
import { Loader, FlexContainerBasic, H3, GridThreeColItem, Grid, Pagination, Section } from '@reapit/elements'
import ErrorBoundary from '@/components/hocs/error-boundary'
import { useHistory, useLocation } from 'react-router'
import AppList from '@/components/ui/app-list'
// Commenting out as we are disabling for launch because there are too few apps
// import AppSidebar from '@/components/ui/app-sidebar'
import { AppDetailState } from '@/reducers/app-detail'
import { selectAppSummary } from '@/selector/client'
import { AppSummaryModel } from '@reapit/foundations-ts-definitions'
import { addQuery, hasFilterParams } from '@/utils/client-url-params'
import Routes from '@/constants/routes'
import { MEDIA_INDEX } from '@/constants/media'
import featureImagePlaceHolder from '@/assets/images/default-feature-image.jpg'
import { Link } from 'react-router-dom'

export const handleAfterClose = ({ setVisible }) => () => setVisible(false)
export const handleOnChange = history => (page: number) => {
  history.push(addQuery({ page }))
}

export interface onCardClickParams {
  setVisible: (isVisible: boolean) => void
  setStateViewBrowse: () => void
  appDetail: AppDetailState
  fetchAppDetail: (id: string, client: string) => void
  clientId: string
}

export const handleOnCardClick = (history: History) => (app: AppSummaryModel) => {
  history.push(`${Routes.CLIENT}/${app.id}`)
}

export const Client: React.FunctionComponent = () => {
  const history = useHistory()
  const location = useLocation()

  const appSummaryState = useSelector(selectAppSummary)
  const urlParams = new URLSearchParams(location.search)
  const page = Number(urlParams.get('page')) || 1

  const hasParams = hasFilterParams(location.search)
  const unfetched = !appSummaryState.data
  const loading = appSummaryState.isAppSummaryLoading
  const apps = appSummaryState?.data?.apps?.data || []
  const featuredApps = appSummaryState?.data?.featuredApps || []
  const { totalCount, pageSize } = appSummaryState?.data?.apps || {}

  return (
    <ErrorBoundary>
      <FlexContainerBasic flexColumn dataTest="page-client-apps-container">
        {/* <AppSidebar /> */}
        {unfetched || loading ? (
          <Loader />
        ) : (
          <>
            <Section>
              <H3 className="mb-0">Browse Apps</H3>
            </Section>
            {!hasParams && featuredApps.length > 0 && (
              <div className="pb-4 bb mb-4">
                <Grid isMultiLine>
                  {featuredApps.map(app => {
                    const featureImageSrc = app?.media?.[MEDIA_INDEX.FEATURE_IMAGE]?.uri || featureImagePlaceHolder
                    return (
                      <GridThreeColItem className="" key={app.id}>
                        <div className="card">
                          <div className="card-image">
                            <Link className="image" to={`${Routes.CLIENT}/${app.id}`}>
                              <img key={app.id} src={featureImageSrc} />
                            </Link>
                          </div>
                        </div>
                      </GridThreeColItem>
                    )
                  })}
                </Grid>
              </div>
            )}
            <AppList
              list={apps}
              loading={loading}
              onCardClick={handleOnCardClick(history)}
              infoType={page > 1 || hasParams ? '' : 'CLIENT_APPS_EMPTY'}
            />
            <Pagination
              totalCount={totalCount}
              pageSize={pageSize}
              pageNumber={page}
              onChange={handleOnChange(history)}
            />
          </>
        )}
      </FlexContainerBasic>
    </ErrorBoundary>
  )
}

export default Client
