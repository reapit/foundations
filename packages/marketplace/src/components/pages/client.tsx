import * as React from 'react'
import { connect } from 'react-redux'
import { ReduxState, FormState } from '@/types/core'
import { ClientState } from '@/reducers/client'
import { Loader } from '@reapit/elements'
import ErrorBoundary from '@/components/hocs/error-boundary'
import { withRouter, RouteComponentProps } from 'react-router'
import AppList from '@/components/ui/app-list'
import AppSidebar from '@/components/ui/app-sidebar'
import { appDetailRequestData } from '@/actions/app-detail'
import { AppDetailState } from '@/reducers/app-detail'
import AppDetailModal from '@/components/ui/app-detail-modal'
import { selectClientId } from '@/selector/client'
import { AppSummaryModel } from '@reapit/foundations-ts-definitions'
import styles from '@/styles/pages/client.scss?mod'
import { appInstallationsSetFormState } from '@/actions/app-installations'
import { addQuery, getParamValueFromPath, hasFilterParams } from '@/utils/client-url-params'
import { setAppDetailModalStateBrowse } from '@/actions/app-detail-modal'

export interface ClientMappedActions {
  setStateViewBrowse: () => void
  fetchAppDetail: (id: string, clientId: string) => void
  installationsSetFormState: (formState: FormState) => void
}

export interface ClientMappedProps {
  clientState: ClientState
  appDetail: AppDetailState
  clientId: string
  installationsFormState: FormState
}

export const handleAfterClose = ({ setVisible }) => () => setVisible(false)
export const handleOnChange = history => (page: number) => {
  history.push(addQuery({ page }))
}

export interface onCardClickParams {
  setVisible: (isVisible: boolean) => void
  setStateViewBrowse: () => void
  appDetail: AppDetailState
  fetchAppDetail: (id: string, client: string) => void
  clientId: string
}

export const handleOnCardClick = ({
  setVisible,
  setStateViewBrowse,
  appDetail,
  fetchAppDetail,
  clientId,
}: onCardClickParams) => (app: AppSummaryModel) => {
  setVisible(true)
  setStateViewBrowse()
  const isCacheEmpty = !appDetail.appDetailData
  const isCacheStale = appDetail.isStale
  const currentAppIdNotMatchWithCachedAppId = appDetail.appDetailData?.data?.id !== app.id

  if (app.id && (isCacheEmpty || isCacheStale || currentAppIdNotMatchWithCachedAppId)) {
    fetchAppDetail(app.id, clientId)
  }
}

export const handleInstallationDone = ({
  isDone,
  installationsSetFormState,
  fetchAppDetail,
  appDetail,
  clientId,
}) => () => {
  if (isDone) {
    installationsSetFormState('PENDING')
    fetchAppDetail(appDetail.appDetailData.data.id, clientId)
  }
}

export type ClientProps = ClientMappedActions & ClientMappedProps & RouteComponentProps<{ page?: any }>

export const Client: React.FunctionComponent<ClientProps> = ({
  clientState,
  history,
  location,
  fetchAppDetail,
  appDetail,
  clientId,
  setStateViewBrowse,
  installationsFormState,
  installationsSetFormState,
}) => {
  const pageNumber =
    !isNaN(Number(getParamValueFromPath(location.search, 'page'))) &&
    Number(getParamValueFromPath(location.search, 'page')) > 0
      ? Number(getParamValueFromPath(location.search, 'page'))
      : 1
  const hasParams = hasFilterParams(location.search)
  const unfetched = !clientState.clientData
  const loading = clientState.loading
  const apps = clientState?.clientData?.apps?.data || []
  const featuredApps = clientState?.clientData?.featuredApps || []
  const { totalCount, pageSize } = clientState?.clientData?.apps || {}
  const [visible, setVisible] = React.useState(false)

  const isDone = installationsFormState === 'DONE'

  React.useEffect(handleInstallationDone({ isDone, installationsSetFormState, fetchAppDetail, appDetail, clientId }), [
    isDone,
  ])

  return (
    <ErrorBoundary>
      <div id="page-client-apps-container" className={styles.clientContainer}>
        <AppSidebar />
        {unfetched || loading ? (
          <Loader />
        ) : (
          <div className={styles.clientContent}>
            {!hasParams && featuredApps.length > 0 && (
              <>
                <AppList
                  list={featuredApps}
                  title="Featured Apps"
                  loading={loading}
                  onCardClick={handleOnCardClick({
                    setVisible,
                    setStateViewBrowse,
                    appDetail,
                    fetchAppDetail,
                    clientId,
                  })}
                  infoType="CLIENT_APPS_EMPTY"
                  numOfColumn={3}
                />
                <div className={styles.divider} />
              </>
            )}
            <AppList
              list={apps}
              loading={loading}
              onCardClick={handleOnCardClick({ setVisible, setStateViewBrowse, appDetail, fetchAppDetail, clientId })}
              infoType={pageNumber > 1 || hasParams ? '' : 'CLIENT_APPS_EMPTY'}
              pagination={{
                totalCount,
                pageSize,
                pageNumber,
                onChange: handleOnChange(history),
              }}
              numOfColumn={3}
            />
          </div>
        )}
      </div>
      <AppDetailModal visible={visible} afterClose={handleAfterClose({ setVisible })} />
    </ErrorBoundary>
  )
}

export const mapStateToProps = (state: ReduxState): ClientMappedProps => ({
  clientState: state.client,
  appDetail: state.appDetail,
  clientId: selectClientId(state),
  installationsFormState: state.installations.formState,
})

export const mapDispatchToProps = (dispatch: any): ClientMappedActions => ({
  setStateViewBrowse: () => dispatch(setAppDetailModalStateBrowse()),
  fetchAppDetail: (id: string, clientId: string) => dispatch(appDetailRequestData({ id, clientId })),
  installationsSetFormState: (formState: FormState) => dispatch(appInstallationsSetFormState(formState)),
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Client))
