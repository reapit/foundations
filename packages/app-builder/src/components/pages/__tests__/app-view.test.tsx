import * as React from 'react'
import { render } from '@testing-library/react'
import { MockedProvider } from '@apollo/client/testing'

import AppView from '../app-view'

import routeData from 'react-router'

describe('AppView', () => {
  const mockParams = {
    appId: '123',
    pageId: '456',
  }
  const mockLocation = {
    pathname: '/123/456',
    hash: '',
    search: '',
    state: '',
  }
  beforeEach(() => {
    jest.spyOn(routeData, 'useParams').mockReturnValue(mockParams)
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation)
  })
  it('should match a snapshot', () => {
    const { asFragment } = render(
      <MockedProvider>
        <AppView />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
