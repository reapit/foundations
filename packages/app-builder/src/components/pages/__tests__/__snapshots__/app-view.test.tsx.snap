// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-93 el-snack-holder"
  />
  <div>
    Not Found
  </div>
</DocumentFragment>
`;
